internal/gke/update.go

```diff
@@ -375,38 +373,27 @@ func UpdateLabels(
 	if config.Spec.Labels == nil || reflect.DeepEqual(config.Spec.Labels, upstreamSpec.Labels) || (upstreamSpec.Labels == nil && len(config.Spec.Labels) == 0) {
 		return NotChanged, nil
 	}
-	logrus.Infof("updating cluster labels for cluster [%s]", config.Name)
-	backoff := wait.Backoff{
-		Duration: 5 * time.Second,
-		Steps:    2,
+	cluster, err := GetCluster(ctx, client, &config.Spec)
+	if err != nil {
+		return NotChanged, err
 	}
-	err := wait.ExponentialBackoff(backoff, func() (bool, error) {
-		cluster, err := GetCluster(ctx, client, &config.Spec)
-		if err != nil {
-			return false, err
-		}
-		_, err = client.Projects.
-			Locations.
-			Clusters.
-			SetResourceLabels(
-				ClusterRRN(config.Spec.ProjectID, Location(config.Spec.Region, config.Spec.Zone), config.Spec.ClusterName),
-				&gkeapi.SetLabelsRequest{
-					LabelFingerprint: cluster.LabelFingerprint,
-					ResourceLabels:   config.Spec.Labels,
-				},
-			).Context(ctx).
-			Do()
-		if err != nil && strings.Contains(err.Error(), "Labels could not be set due to fingerprint mismatch") {
-			logrus.Debug("retrying label update")
-			return false, nil
-		}
-		if err != nil {
-			logrus.Debugf("error during label update: %v", err)
-			return false, err
-		}
-		return true, nil
-	})
+	logrus.Infof("updating cluster labels for cluster [%s]", config.Name)
+	_, err = client.Projects.
+		Locations.
+		Clusters.
+		SetResourceLabels(
+			ClusterRRN(config.Spec.ProjectID, Location(config.Spec.Region, config.Spec.Zone), config.Spec.ClusterName),
+			&gkeapi.SetLabelsRequest{
+				LabelFingerprint: cluster.LabelFingerprint,
+				ResourceLabels:   config.Spec.Labels,
+			},
+		).Context(ctx).
+		Do()
 	if err != nil {
+		if strings.Contains(err.Error(), "Labels could not be set due to fingerprint mismatch") {
+			logrus.Debugf("got transient error: %v, retrying", err)
+			return Retry, nil
```

Comment:
`Retry` is being returned here with no error. However, the caller doesn't seem to check for `Retry` and if there is no error then it seems that this call will not be retried.

Reply:
fixed